# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/Notification.h
# Commit: bb63792c3333
# Full Hash: bb63792c3333a7911bdfad9e9e9ec58bb48bad93
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 2: Create PNotification via PBackground r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227687
# ==============================================================================

diff -r 93937caf6d13 -r bb63792c3333 dom/notification/Notification.h
--- a/dom/notification/Notification.h	Thu Nov 07 17:32:03 2024 +0000
+++ b/dom/notification/Notification.h	Thu Nov 07 17:32:03 2024 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/GlobalFreezeObserver.h"
 #include "mozilla/UniquePtr.h"
 #include "mozilla/dom/NotificationBinding.h"
+#include "mozilla/dom/notification/NotificationChild.h"
 #include "mozilla/dom/WorkerPrivate.h"
 #include "mozilla/dom/quota/QuotaCommon.h"
 
@@ -273,6 +274,8 @@
     mScope = aScope;
   }
 
+  WeakPtr<notification::NotificationChild> mActor;
+
   const nsString mID;
   const nsString mTitle;
   const nsString mBody;
@@ -318,6 +321,8 @@
       const NotificationOptions& aOptions, const nsAString& aScope,
       ErrorResult& aRv);
 
+  bool CreateActor(Promise* aPromise);
+
   nsIPrincipal* GetPrincipal();
 
   nsresult PersistNotification();